_episodes/29.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: episode
+title:  "29: What is Trino and the Hive connector"
+date: 2021-10-28
+tags: beginner sql hive connector
+youtube_id: "xAnKbpKSPtA"
+wistia_id: "75w1kryofe"
+sections: 
+   - title: "Concept of the week"
+     desc: "What is Trino?"
+     time: 1023
+   - title: "PR of the week"
+     desc: "PR 8821 Add HTTP/S query event logger"
+     time: 3515
+   - title: "Question of the week"
+     desc: "Does the Hive connector depend on the Hive runtime?"
+     time: 3763
+---
+
+## Release 364 
+
+Release 364 is just around the corner, here is Manfred's release preview:
+
+* ALTER MATERIALIZED VIEW ... RENAME TO
+* whole bunch of performance improvements
+* Elasticsearch connector no longer fails with unsupported types
+* Hive connector has optimize procedure now!
+* Parquet and avro fixes and improvements
+
+## Concept of the week: What is Trino?
+
+Trino is the project created by Martin Traverso, Dain Sundstrom, David Phillips,
+and Eric Hwang in 2012 to replace the 300PB Hive data warehouse at Facebook. The
+goal of Trino is to run fast ad-hoc analytics queries over big data file systems
+like HDFS and object stores like S3. 
+
+An initially unintended but now characteristic feature of Trino is also its 
+ability to execute federated queries over various distributed data sources. This
+includes, but is not limited to: Accumulo, BigQuery, Apache Cassandra, 
+ClickHouse, Druid, Elasticsearch, Google Sheets, Apache Iceberg, Apache Hive, 
+JMX, Apache Kafka, Kinesis, Kudu, MongoDB, MySQL, Oracle, Apache Phoenix, 
+Apache Pinot, PostgreSQL, Prometheus, Redis, Redshift, SingleStore (MemSQL), 
+Microsoft SQL Server. 
+ 
+How does Trino query across everything from data lakes, SQL, and NoSQL databases
+at unprecedented speeds? It will help to start by going over Trino's 
+architecture to understand the answer to that question.
+
+<p align="center">
+<img align="center" width="75%" height="100%" src="/assets/episode/29/1-architecture.png"/><br/>
+Source: <a href="https://trino.io/blog/2021/04/21/the-definitive-guide.html">Trino: The Definitive Guide</a>.
+</p>
+
+Trino consists of two types of nodes, a _coordinator_ and _worker_ nodes. The 
+coordinator parses, analyzes, plan, and schedules SQL queries input by the user.
```

Comment:
plans, and schedules the processing of SQL queries. The queries are submitted by users directly or with connected SQL reporting tools.